packages/typespec-vscode/src/emit/emit.ts

```diff
@@ -0,0 +1,395 @@
+    if (compileResult.exitCode !== 0) {
+      logger.error(
+        `Failed to generate ${selectedEmitter.emitterKind} code for ${selectedEmitter.language}.`,
+        [],
+        {
+          showOutput: true,
+          showPopup: true,
+          progress: overallProgress,
+        },
+      );
+    } else {
+      logger.info(
+        `Generating ${selectedEmitter.emitterKind} code for ${selectedEmitter.language}...Succeeded`,
+        [],
+        {
+          showOutput: true,
+          showPopup: true,
+          progress: overallProgress,
+        },
+      );
+    }
+  } catch (err) {
+    logger.error(
+      `Exception occurred when generating ${selectedEmitter.emitterKind} code for ${selectedEmitter.language}.`,
+      [err],
+      {
+        showOutput: true,
+        showPopup: true,
+        progress: overallProgress,
+      },
+    );
+  }
+
+  /*TODO: build sdk. */
+}
+
+export async function emitCode(
+  context: vscode.ExtensionContext,
+  uri: vscode.Uri,
+  overallProgress: vscode.Progress<{ message?: string; increment?: number }>,
+) {
+  let tspProjectFile: string = "";
+  if (!uri) {
+    const targetPathes = await TraverseMainTspFileInWorkspace();
+    logger.info(`Found ${targetPathes.length} ${StartFileName} files`);
+    if (targetPathes.length === 0) {
+      logger.info("No main tsp file found. Generating Cancelled.", [], {
+        showOutput: false,
+        showPopup: true,
+        progress: overallProgress,
+      });
+      return;
+    }
+    const toProjectPickItem = (filePath: string): TypeSpecProjectPickItem => {
+      return {
+        label: filePath,
+        path: filePath,
+        iconPath: {
+          light: Uri.file(context.asAbsolutePath(`./icons/tsp-file.light.svg`)),
+          dark: Uri.file(context.asAbsolutePath(`./icons/tsp-file.dark.svg`)),
+        },
+      };
+    };
+    const typespecProjectQuickPickItems: TypeSpecProjectPickItem[] = targetPathes.map((filePath) =>
+      toProjectPickItem(filePath),
+    );
+    const selectedProjectFile = await vscode.window.showQuickPick(typespecProjectQuickPickItems, {
+      title: "Select a TypeSpec Project",
+      canPickMany: false,
+      placeHolder: "Pick a project",
+      ignoreFocusOut: true,
+    });
+    if (!selectedProjectFile) {
+      logger.info("No project selected. Generating Cancelled.", [], {
+        showOutput: false,
+        showPopup: true,
+        progress: overallProgress,
+      });
+      return;
+    }
+    tspProjectFile = selectedProjectFile.path;
+  } else {
+    const tspStartFile = await getMainTspFile(uri.fsPath);
+    if (!tspStartFile) {
+      logger.info("No main file. Invalid typespec project.", [], {
+        showOutput: false,
+        showPopup: true,
+        progress: overallProgress,
+      });
+      return;
+    }
+    tspProjectFile = tspStartFile;
+  }
+
+  interface EmitTypeQuickPickItem extends vscode.QuickPickItem {
+    emitterKind: EmitterKind;
+  }
+
+  const codesToEmit = [
```

Comment:
can we parse from config instead of hard code here?

Reply:
updated.